### ISSUE TITLE
AVT: Accessibility | Screen Reader Understandable Issue - No data Information 

### ISSUE THREAD

**Issue by kumarivijeta** [carbon-design-system/carbon-components-angular#2844]
## Detailed description

> Describe in detail the issue you're having.
The screen reader is unable to read the non sortable table headers  and the No data information for an empty table.
For a person who is visually impaired it would be difficult to know the information in context.

> Is this a feature request (new component, new icon), a bug, or a general issue?
**Bug**

> Is this issue related to a specific component?

<cds-table>

> What did you expect to happen? What happened instead? What would you like to see changed?

The screen reader should read out the information in context

> What browser are you working in?

Chrome, Edge

> What version of the Carbon Design System are you using?

carbon-components-angular@5.14.1

## Steps to reproduce the issue

1. set sortable false , Open screen reader and observe

## Additional information

* Screenshots or code
<img width="1083" alt="image" src="https://github.com/carbon-design-system/carbon-components-angular/assets/60390189/edcebe20-57da-4cf5-a12a-dc041137570b">
